### PR TITLE
updated VPC creation file with count and dynamic count

### PR DESCRIPTION
**Pull request by sinhprab**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,3 +1,10 @@
+locals {
+  public_cidr       = ["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr      = ["10.0.2.0/24", "10.0.3.0/24"]
+  availability_zone = ["eu-west-2a", "eu-west-2b"]
+
+}
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  availability_zone = ["eu-west-2a", "eu-west-2b"]
}
```

File: main.tf [hunk 2]
```diff
@@ -93,48 +75,33 @@ resource "aws_route_table" "public" {
...
+  count = length(local.private_cidr)
 
-resource "aws_route_table_association" "private02" {
-  subnet_id      = aws_subnet.private02.id
-  route_table_id = aws_route_table.private02.id
+  subnet_id      = aws_subnet.private[count.index].id
+  route_table_id = aws_route_table.private[count.index].id
 }
+
+
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
}
```

File: main.tf [hunk 3]
```diff
@@ -49,37 +42,26 @@ resource "aws_internet_gateway" "main" {
...
-  tags = {
-    Name = "nat01"
-  }
-}
+resource "aws_eip" "nat" {
+  count = length(local.public_cidr)
 
-resource "aws_eip" "nat02" {
   vpc = true
   tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  vpc = true
  
  tags = {
```

File: main.tf [hunk 4]
```diff
@@ -6,39 +13,25 @@ resource "aws_vpc" "main" {
...
+resource "aws_subnet" "private" {
+  count = length(local.private_cidr)
 
-resource "aws_subnet" "private02" {
   vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.3.0/24"
-  availability_zone = "eu-west-2b"
+  cidr_block        = local.private_cidr[count.index]
+  availability_zone = local.availability_zone[count.index]
   tags = {
```
Comment (vladimirmukhin) [inline main.tf]:
```suggestion
  availability_zone = local.availability_zone[count.index]
  
  tags = {
```

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (sinhprab) [reply main.tf]: Done , removed  the extra line

File: main.tf (hunk 2 above)
Comment (sinhprab) [reply main.tf]: Done , deleted  the extra line

File: main.tf (hunk 3 above)
Comment (sinhprab) [reply main.tf]: Done , added  the extra line

File: main.tf (hunk 4 above)
Comment (sinhprab) [reply main.tf]: Done , added  the extra line

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**